src/views/Profile/AdminTable/AdminDatePicker.tsx

```diff
@@ -54,39 +132,51 @@ export const AdminDatePicker: React.FC<IMaterialsDate> = ({
           value={[]}
           disableUnderline
           displayEmpty
-          renderValue={() => (
-            <div>
-              {t(langTokens.admin.from)} {parsDate(start)}{' '}
-              {t(langTokens.admin.to)} {parsDate(end)}
-            </div>
-          )}
+          renderValue={() =>
+            !datePickerState.firstSelectedDay.value &&
+            !datePickerState.secondSelectedDay.value ? (
+              <DateRangeOutlinedIcon />
+            ) : (
+              <div>
+                {t(langTokens.admin.filterByDatePeriod, {
+                  rangeStart: parsDate(start),
+                  rangeEnd: parsDate(end),
+                })}
+              </div>
+            )
+          }
           MenuProps={MenuProps}
         >
           <Box display="flex" className={classes.pickersWrapper}>
-            <DatePicker
-              autoOk
-              orientation="portrait"
-              variant="static"
-              openTo="date"
-              value={start}
-              onChange={(value) => {
-                handleDatePicker('start', value);
-              }}
-              format="dd-mm-yyyy"
-              maxDate={end}
-            />
-            <DatePicker
-              autoOk
-              orientation="portrait"
-              variant="static"
-              openTo="date"
-              value={end}
-              onChange={(value) => {
-                handleDatePicker('end', value);
-              }}
-              format="dd-mm-yyyy"
-              minDate={start}
-            />
+            <Box className={datepickerClassesStart.datePicker}>
```

Comment:
I think, instead of putting color into a reducer and then create 2 different `datepickerClassesStart` and `datepickerClassesEnd` we could instead:

```
  const datepickerClassesSelectedDate = useStylesForDatePicker({
    bgColorForDatePicker: 'grey',
  });
  const datepickerClassesNoSelection = useStylesForDatePicker({
    bgColorForDatePicker: 'transparent',
  });
......
<Box className={start ? datepickerClassesSelectedDate.datePicker :  datepickerClassesNoSelection.datePicker }>
....
<Box className={end ? datepickerClassesSelectedDate.datePicker :  datepickerClassesNoSelection.datePicker }>
```

A bit repetitive? may be. but we don't have any other difference between Start Date and End Date selector, we just need to suppress current day highlighted, right?

Also this would allow us to get rid of extra state(`useReducer` it's also a state) and consume `end`/`start` again.

--------

Ideally, I'd like to have `DokazoviDatePicker.tsx` which would implement logic under the hood:

1. no highlighting for the current day if no selection
2. clicking currently selected day unselects it

This way we would re-use it across the app getting unified UX.

Do we use DatePicker elsewhere?

Reply:
We use DatePicker in SetPublicationDateModal. But not use AdminDatePicker.

Reply:
do we expect current day to be highlighted in SetPublicationDateModal?

Reply:
No, I asked Denis and he said that we need there the same behavior as this.

Reply:
understood. Then I propose to extract this into a separate component - also `<DatePicker` but which will have all this extra styling. All extra props should be passed down to `material-ui/pickers`' DatePicker. Then replacing it will just a change in `import .. from ` line

Easier to reuse, no risks of inconsistent changes

Reply:
so will you extract this into a separate component, so reducer here will not orchestrate colors?

Reply:
this is what I thought about:

```
import { DatePicker as MuiDatePicker } from '@material-ui/pickers';
import {createMuiTheme, makeStyles} from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'contents',
    '& .MuiPickersDay-daySelected': {
      backgroundColor: ({ isHighlightingEnabled }) => isHighlightingEnabled ? theme.palette.primary.main : 'transparent',
      color: ({ isHighlightingEnabled }) => isHighlightingEnabled ? theme.palette.primary.contrastText : theme.palette.text.primary
    }
  }
}))

export function DatePicker(props) {
  const isHighlightingEnabled = !!props.value;
  const classes = useStyles({ isHighlightingEnabled })
  const onChange = useCallback((value) => {
    if (String(value) === String(props.value)) {
      props.onChange(null)
    } else {
      props.onChange(value)
    }
  }, [props.value, props.onChange])

  return (<div className={classes.container}>
    <MuiDatePicker 
      {...props} 
      variant="static"
      onChange={onChange} 
    />
  </div>)
}
```

and then instead of `import { DatePicker } from '@material-ui/pickers'` we could `import { DatePicker } from '../ourCustomDatePicker.tsx'` and use it just like if `material-ui`'s version would patched and worked with "unsetting" date and suppressing highlighting today day if `value` is `null`

the big disadvantage of that method - styles will work fine only if `variant="static"`, otherwise Dialog is rendered outside of our "extra div with custom class" and override does not work

What do you say?